From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RedeObscurity <zmonsterpotente@gmail.com>
Date: Tue, 28 Dec 2021 14:48:49 -0300
Subject: [PATCH] (Obscurity) Skip cloning loot parameters

Original code by RedeObscurity, licensed under MIT
You can find the original code on https://github.com/RedeObscurity/Obscurity

diff --git a/src/main/java/net/minecraft/advancements/Advancement.java b/src/main/java/net/minecraft/advancements/Advancement.java
index 2f3d9e5b849093027d3c2ef434494cd600f52a31..ee07b1f6fb32697c9b08a770156b72f5560d6c9d 100644
--- a/src/main/java/net/minecraft/advancements/Advancement.java
+++ b/src/main/java/net/minecraft/advancements/Advancement.java
@@ -44,7 +44,7 @@ public class Advancement {
     public Advancement(MinecraftKey minecraftkey, @Nullable Advancement advancement, @Nullable AdvancementDisplay advancementdisplay, AdvancementRewards advancementrewards, Map<String, Criterion> map, String[][] astring) {
         this.key = minecraftkey;
         this.display = advancementdisplay;
-        this.criteria = ImmutableMap.copyOf(map);
+        this.criteria = java.util.Collections.unmodifiableMap(map); // Obscurity - Skip cloning loot parameters
         this.parent = advancement;
         this.rewards = advancementrewards;
         this.requirements = astring;
diff --git a/src/main/java/net/minecraft/core/RegistryMaterials.java b/src/main/java/net/minecraft/core/RegistryMaterials.java
index 18f78a5cb03dd9c8349c28b99d013752e32c9167..6d9f0b4f4d201cbeec3cd4b9e021fd9532cb1efe 100644
--- a/src/main/java/net/minecraft/core/RegistryMaterials.java
+++ b/src/main/java/net/minecraft/core/RegistryMaterials.java
@@ -238,7 +238,7 @@ public class RegistryMaterials<T> extends IRegistryWritable<T> {
             });
             return registrymaterials;
         }, (registrymaterials) -> {
-            return ImmutableMap.copyOf(registrymaterials.bi);
+            return java.util.Collections.unmodifiableMap(registrymaterials.bi); // Obscurity - Skip cloning loot parameters
         });
     }
 
